src/workerd/api/global-scope.h

```diff
@@ -95,7 +95,7 @@ class PromiseRejectionEvent: public Event {
   void visitForGc(jsg::GcVisitor& visitor);
 };
 
-class WorkerGlobalScope: public EventTarget {
+class WorkerGlobalScope: public EventTarget, public jsg::ContextGlobal {
```

Comment:
I believe this ends up inheriting two copies of `jsg::Object`, one via `EventTarget` and one via `jsg::ContextGlobal`. Since it's not virtual inheritance, these two copies will actually be two separate instances.

This worries me. All of the `jsg::Wrapper` machinery will be duplicated, presumably with just one of the two copies actually being used. But if you aren't careful you could easily use the other copy by accident. I'm actually a bit surprised this isn't causing ambiguity errors as-is, with the compiler not knowing which instance you intended to use...

I think it might be necessary to change `jsg::ContextGlobal` such that it does not inherit `jsg::Object`. Then, all classes that want to impalement a context must inherit both `jsg::Object` and `jsg::ContextGlobal`.

Reply:
Oh wait! I was confused between `ContextGlobal` and `ContextGlobalObject`.